src/k8s/pkg/k8sd/app/cluster_util.go

```diff
@@ -0,0 +1,130 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path"
+
+	"github.com/canonical/k8s/pkg/k8sd/api/impl"
+	"github.com/canonical/k8s/pkg/k8sd/pki"
+	"github.com/canonical/k8s/pkg/k8sd/setup"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/k8s/pkg/utils/k8s"
+	"github.com/canonical/microcluster/state"
+)
+
+func setupDatastoreCertificates(snap snap.Snap, cfg types.ClusterConfig, nodeName string, allowSelfSignedCA bool) (error, *pki.K8sDqlitePKI, *pki.ExternalDatastorePKI) {
+	// Certificates
+	switch cfg.APIServer.Datastore {
+	case "k8s-dqlite":
+		dqliteCert := pki.NewK8sDqlitePKI(pki.K8sDqlitePKIOpts{
+			Hostname:          nodeName,
+			IPSANs:            []net.IP{{127, 0, 0, 1}},
+			Years:             20,
+			AllowSelfSignedCA: allowSelfSignedCA,
+		})
+
+		if err := dqliteCert.CompleteCertificates(); err != nil {
+			return fmt.Errorf("failed to initialize cluster certificates: %w", err), nil, nil
```

Comment:
let's update the error messages here (and in other places) to reflect what the code is trying to do

```suggestion
			return fmt.Errorf("failed to initialize k8s-dqlite certificates: %w", err), nil, nil
```

Reply:
I was beginning to think the same, this step looks the same on the first glance but actually isn't.